go/master/client.go

```diff
@@ -15,16 +17,58 @@ type Addresser interface {
 // Client is the client of the master server.
 type Client struct {
 	conn *connection.Conn
+	ch   chan []byte
 }
 
 // NewClient creates a new Client.
 func NewClient(addr Addresser) *Client {
 	c := &Client{}
 	c.conn = connection.New()
+	c.ch = make(chan []byte)
 	go c.monitorMaster(addr)
+	go c.getRecords()
 	return c
 }
 
+func (c *Client) getRecords() {
+	for {
+		t, err := c.getTask()
+		if err != nil {
+			// TODO(helin): wait before move on with next
+			// getTask call.
+			log.Println(err)
+			continue
+		}
+
+		for _, chunk := range t.Chunks {
+			f, err := os.Open(chunk.Path)
+			if err != nil {
+				log.Println(err)
+				continue
+			}
+
+			s := recordio.NewRangeScanner(f, &chunk.Index, -1, -1)
+			for s.Scan() {
+				c.ch <- s.Record()
```

Comment:
We can use [`paddle.reader.buffered`](https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/v2/reader/decorator.py#L162) for this purpose.
The good thing about having an unbuffered channel is we can get most accurate report for task finish. Imagine if we have use buffered channel here with buffer size 1000, the task finish will be 1000 data instance earlier.